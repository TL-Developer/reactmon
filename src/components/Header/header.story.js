'use strict';

import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { linkTo } from '@storybook/addon-links';
import Header from './index'

const stories = storiesOf('Header', module)

stories.add('first story', () => (
  <Header />
))
