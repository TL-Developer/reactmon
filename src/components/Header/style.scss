.CcHeader {
  background: #db4242;
  width: 100%;
  height: 50px;
  overflow: hidden;

  h3 {
    text-align: center;
    color: #fff;
    padding-top: 1.5%;
  }

  .app-title {
    position: absolute;
    right: 5%;
    top: 1.5%;

    h3 {
      margin: 0;
      padding: 0;
    }
  }

  ul.list {
    list-style: none;
    margin: 0;
    padding: 0;
    height: 100%;

    li {
      padding: 1.4% 2%;
      float: left;
      height: 100%;
      cursor: pointer;
      -webkit-transition: all .5s ease;
      -moz-transition: all .5s ease;
      -ms-transition: all .5s ease;
      -o-transition: all .5s ease;
      transition: all .5s ease;
      color: #fff;

      &:hover {
        background: #fff;
        color: #db4242;
      }

      &:hover {
        border-top: 0;
        border-left: 5px;
        border-right: 0px;
        border-bottom: 0;
        border-style: solid;
        border-color: #ff6969;
      }
    }
  }
}
