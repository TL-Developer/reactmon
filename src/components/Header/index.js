import React, { Component } from 'react'
import PropTypes from 'prop-types';
import './style.scss'

const Header = ({slogan}) => (
  <header className="CcHeader">
    <ul className="list">
      <li route="/">Home</li>
      <li route="/about">About</li>
      <li>
        <a href="https://github.com/TL-Developer/PokeES6" target="_black">Github</a>
      </li>
    </ul>
    <div className="app-title">
      <h3>{slogan}</h3>
    </div>
  </header>
);

Header.proptypes = {
  slogan: PropTypes.string.isRequired
}

export default Header;
