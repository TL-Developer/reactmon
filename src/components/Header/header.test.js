'use strict';

import React from 'react';
import renderer from 'react-test-renderer';
import Header from './index';

it('Should Header to match snapshot', () => {
  const tree = renderer.create(<Header slogan={"Slogan"} />).toJSON();
  expect(tree).toMatchSnapshot();
});
