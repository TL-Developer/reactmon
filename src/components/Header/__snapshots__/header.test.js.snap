// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Header to match snapshot 1`] = `
<header
  className="CcHeader"
>
  <ul
    className="list"
  >
    <li
      route="/"
    >
      Home
    </li>
    <li
      route="/about"
    >
      About
    </li>
    <li>
      <a
        href="https://github.com/TL-Developer/reactmon"
        target="_black"
      >
        Github
      </a>
    </li>
  </ul>
  <div
    className="app-title"
  >
    <h3>
      Slogan
    </h3>
  </div>
</header>
`;
